sbtc-core/src/signer/mod.rs

```diff
@@ -0,0 +1,119 @@
+/// An sBTC signer configuration
+pub mod config;
+
+use std::collections::HashMap;
+
+use crate::signer::config::Config;
+use crate::SBTCError;
+use bitcoin::{Network, PrivateKey, PublicKey, Transaction as BitcoinTransaction};
+use p256k1::ecdsa;
+use url::Url;
+use wsts::{bip340::SchnorrProof, common::Signature};
+
+#[derive(Default, Clone, Debug)]
+pub struct SignersData {
+    pub signers: HashMap<u32, ecdsa::PublicKey>,
+    pub vote_shares: HashMap<u32, ecdsa::PublicKey>,
+}
+
+/// Coordinator algorithm trait
+pub trait CoordinatorAlgorithm {
+    fn generate_sbtc_wallet_public_key(&self) -> Result<PublicKey, SBTCError>;
+    fn run_signing_round(
+        &self,
+        _tx: BitcoinTransaction,
+    ) -> Result<(Signature, SchnorrProof), SBTCError>;
+}
+
+/// Signer algorithm trait
+pub trait SignerAlgorithm: CoordinatorAlgorithm {
+    fn sign_message(&self, message: &[u8]) -> Result<Vec<u8>, SBTCError>;
```

Comment:
Accepting a mutable buffer would prevent any unnecessary copies, but might not be viable if size changes dramatically.

`fn sign_message(&self, message: &mut [u8]) -> SBTCResult<()>`

Reply:
I don't think accepting a mutable buffer here would help much given the sign_msg functions that seem to be used throughout the crypt side where we use a hash function to create a mutable nbuffer and return that. Wouldbe hard to avoid this copy.

Reply:
Why would the message ever be mutated?

Reply:
I didn't mean mutating the message but from the bdk it seems to me like it keeps an unsigned transaction under the hood, and just modifies it when it signs it. That's why I thought a buffer might come in handy here.

Reply:
Ah yeah I guess the low-level signer API is about creating a signature for arbitrary bytes. A high-level API that adds a signature to a transaction could mimic this.